docs/signing-spec.md

```diff
@@ -342,6 +343,236 @@ For example, if a package is signed by an unknown host key, then this package ha
+
+Note that even in such an attack, the compromised host cannot alter a bindle's content without invalidating the other signatures.
+
+A fully compromised Bindle server is vulnerable to a freeze attack for the obvious reason that the attacker would control all avenues for publishing a Bindle via that host.
+
+A proxy _could_ perpetrate something like a freeze attack if an implementation's only strategy for checking for an update is to contact the proxy.
+Then a proxy can effectively fail to fetch a bindle, and consequently cause a freeze.
+Note, though, that this is a desirable effect of a system such as Bindle, in which an internal proxy may only have a subset of the packages available on the open web.
+Consequently, mitigation of this issue is left to higher-level implementations that may specify that when they get a "new version" (which is a discrete bindle from the "old version"), it follows a strategy proper to its usage.
+
+## Generating a Known-invalid Signature
+
+This section is non-normative and may be removed.
+
+In the case where an existing package is discovered to have a flaw that should prevent its verification, it is possible (though not always recommended, and never required) to generate an intentionally bad signature to force a verification failure.
+
+The recommended way of doing this is to include the `INVALID` string inside of the hash contents after the timestamp.
+This makes it possible for an agent to reconstruct the signature and determine whether the signature was intentionally invalidated.
+
+```
+Matt Butcher <matt.butcher@example.com>
+mybindle
+0.1.0
+creator
+1611960337
+INVALID
+~
+e1706ab0a39ac88094b6d54a3f5cdba41fe5a901
+098fa798779ac88094b6d54a3f5cdba41fe5a901
+5b992e90b71d5fadab3cd3777230ef370df75f5b
+```
+
+Note that simply generating an invalid signature does not necessarily mean that agents will respect this signature.
+An implementation may only mind invalid signatures on keys that are in its keyring or that are in a certain role.
+
+## Key Rotation
+
+Bindle makes no assumptions about the frequency of key rotation.
+However, there are a few non-normative recommendations:
+
+- If a `host` key is rotated, all packages SHOULD be re-signed by the new host key.
+    - For a properly configured Bindle server, this does not introduce an opportunity for an attacker to modify packages because the `creator` and all `proxy` and `approver` signatures remain unaltered, and any attempt to modify the package will invalidate those signatures
+- When a creator rotates keys, and the cause is NOT compromise, the user SHOULD re-sign existing bindles using the new key and the `approver` role. The original `creator` signature SHOULD NOT be altered in any way.
```

Comment:
I could be misremembering, but isn't a bindle only allowed to have one `creator` signature? If I am a consumer, and I see two creator signatures, what does that mean? I can see someone else asserting they are a creator when they aren't

Reply:
The text says they should sign as an `approver` and not alter the `creator` role. So the same _person_ would sign the same bindle two different times... but with two different keys (the old and the new) at two different times.